### PR TITLE
Add doxygen feature for cxx-build 

### PR DESCRIPTION
**Pull request by damoncro**
Close:
- #669

Solution:
Add `doxygen` feature for `cxx-build`:
Enabling this feature will write `///` comments instead of `//` into generated cpp files.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -25,7 +25,7 @@ std = ["alloc"]
 
 [dependencies]
 cxxbridge-macro = { version = "=1.0.72", path = "macro" }
-link-cplusplus = "1.0"
+link-cplusplus = { version = "1.0", features = ["libc++"] }
```
Comment (dtolnay) [inline Cargo.toml:28]: Please remove this unrelated change from the PR.

File: gen/src/write.rs [hunk 2]
```diff
@@ -236,13 +236,21 @@ fn write_struct<'a>(out: &mut OutFile<'a>, strct: &'a Struct, methods: &[&Extern
     writeln!(out, "#ifndef {}", guard);
     writeln!(out, "#define {}", guard);
     for line in strct.doc.to_string().lines() {
-        writeln!(out, "//{}", line);
+        if cfg!(feature = "doxygen") {
```
Comment (dtolnay) [inline gen/src/write.rs:239]: Could you justify why not to always emit `///`? Is it because that would break anyone currently running Doxygen on cxx-generated headers in which the Rust comments are not "valid Doxygen" in some way?

File: gen/src/write.rs [hunk 3]
```diff
@@ -391,14 +411,18 @@ fn write_enum<'a>(out: &mut OutFile<'a>, enm: &'a Enum) {
...
+            writeln!(out, "  //{}", line);
+        }
     }
     write!(out, "enum class {} : ", enm.name.cxx);
     write_atom(out, repr);
     writeln!(out, " {{");
     for variant in &enm.variants {
         for line in variant.doc.to_string().lines() {
-            writeln!(out, "  //{}", line);
+            writeln!(out, "  ///{}", line);
```
Comment (dtolnay) [inline gen/src/write.rs:425]: Missing `if`

File: gen/src/write.rs [hunk 4]
```diff
@@ -337,7 +349,11 @@ fn write_opaque_type<'a>(out: &mut OutFile<'a>, ety: &'a ExternType, methods: &[
     writeln!(out, "#ifndef {}", guard);
     writeln!(out, "#define {}", guard);
     for line in ety.doc.to_string().lines() {
-        writeln!(out, "//{}", line);
+        if cfg!(feature = "doxygen") {
+            writeln!(out, "  ///{}", line);
```
Comment (dtolnay) [inline gen/src/write.rs:353]: Wrong indentation.

Elsewhere as well.

File: gen/src/write.rs [hunk 5]
```diff
@@ -236,13 +236,21 @@ fn write_struct<'a>(out: &mut OutFile<'a>, strct: &'a Struct, methods: &[&Extern
     writeln!(out, "#ifndef {}", guard);
     writeln!(out, "#define {}", guard);
     for line in strct.doc.to_string().lines() {
-        writeln!(out, "//{}", line);
+        if cfg!(feature = "doxygen") {
+            writeln!(out, "  ///{}", line);
```
Comment (dtolnay) [inline gen/src/write.rs:240]: According to https://www.doxygen.nl/manual/docblocks.html, `///` is only considered Doxygen if there are at least 2 of them on consecutive lines. That's not how Rust `///` works normally so it would make sense for cxx-build to ensure it's converting Rust docs to Doxygen docs.

**Review by dtolnay — CHANGES_REQUESTED** [review]

File: gen/src/write.rs (hunk 5 above)
Comment (damoncro) [reply gen/src/write.rs:240]: This PR is not a complete implementation for generating the Doxygen docs. But a small enhancement that is enough for fullfilling the minimal doxygen docs generate requirement.

Two more lines are simple, just input two more lines in rust side.

File: gen/src/write.rs (hunk 4 above)
Comment (damoncro) [reply gen/src/write.rs:353]: What do you mean `wrong indentation`, the indentation is already checked by `rustfmt`.

File: gen/src/write.rs (hunk 3 above)
Comment (damoncro) [reply gen/src/write.rs]: Added

File: gen/src/write.rs (hunk 2 above)
Comment (damoncro) [reply gen/src/write.rs:239]: This PR is not a complete implementation for generating the Doxygen docs. But a small enhancement that is enough for fullfilling the minimal doxygen docs generate requirement. If we need more advance docs, maybe it is worth to add more features later on.

File: gen/src/write.rs (hunk 5 above)
Comment (dtolnay) [reply gen/src/write.rs:240]: I understand that it could be simple for someone to work around, but I am not interested in accepting this PR as currently implemented. Taking Rust docs and turning them into either doxygen docs or silently into something that _looks_ like doxygen docs, but isn't, is going to come across as clearly broken behavior to anybody that uses this.

File: gen/src/write.rs (hunk 4 above)
Comment (dtolnay) [reply gen/src/write.rs:353]: In the generated code, I want the number of spaces in front of the comment to be equal to the number of spaces in front of the thing that it is documenting, like it was prior to this PR.

File: gen/src/write.rs (hunk 5 above)
Comment (damoncro) [reply gen/src/write.rs:240]: Noted, let's find a better way to improve it later~
